app/services/authentication_token_service.rb

```diff
@@ -0,0 +1,14 @@
+class AuthenticationTokenService
+  HMAC_SECRET = 'my$ecretK3y'.freeze
+  ALGORITHM_TYPE = 'HS256'.freeze
+
+  def self.call(user_id)
+    payload = { user_id: }
+
+    JWT.encode payload, HMAC_SECRET, ALGORITHM_TYPE
+  end
+
+  def self.decode(token)
+    JWT.decode token, HMAC_SECRET, true, { algorithm: ALGORITHM_TYPE }[0]
+  end
+end
```

Comment:
When I try to validate the token using the 'GET' /authenticate, I get this error

![image](https://user-images.githubusercontent.com/44676055/179800418-374d60a4-e63d-410c-b2fb-d4c7db736055.png)

I passed the token as a header 'Authorization'. I checked the controller, and it does register and pick up the token but it seems the decode function in the AuthenticationServiceGives this error. Are you able to look into it? 

![image](https://user-images.githubusercontent.com/44676055/179800556-295f7677-d40a-424c-8748-5a344ac716b5.png)

Reply:
thank you @Beardless-sheik for pointing that out, I'll work on it in due time